### ISSUE TITLE
Enhancement: [explicit-member-accessibility] assigning an accessibility level to decorators

### ISSUE THREAD

**Issue by certainlyakey** [typescript-eslint/typescript-eslint#10820]
### Before You File a Proposal Please Confirm You Have Done The Following...

- [x] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22enhancement%3A+plugin+rule+option%22) and found none that match my proposal.
- [x] I have searched the [current rule list](https://typescript-eslint.io/rules/#rules) and found no rules that match my proposal.
- [x] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### My proposal is suitable for this project

- [x] I believe my proposal would be useful to the broader TypeScript community (meaning it is not a niche proposal).

### Link to the rule's documentation

https://typescript-eslint.io/rules/explicit-member-accessibility/#accessibility

### Description

This proposal has already appeared as a [Github issue](https://github.com/typescript-eslint/typescript-eslint/issues/4719) and was closed, however I believe it is applicable to a wider range of uses than mentioned there. At least in 3 major frameworks property/method decorators are used: Angular, Lit, Stencil. 

As mentioned in another ticket, it would be good, while applying the rule to other properties/methods, always allow ones with specific decorators to either pass or fail, according to a project-specific rule config. 

I myself am working on a large Lit library, and applying the rule to it with a support to whitelist certain decorators would be tremendous help. 

### Fail

```TypeScript
@Event('bp-click') public onClick<CustomEvent>; // implicitly considered private by the plugin, as specified in eslint config
```

### Pass

```TypeScript
@Prop() disabled = false; // implicitly considered public by the plugin, as specified in eslint config
```

**Comment by bradzacher** [comment]
I'm not quite sure I understand the ask, fully.

You're saying that certain decorators should imply certain access modifiers and so members with said decorator(s) should be excluded from the rule?

I don't really see the broad applicability here. And I don't quite understand why you'd want to exclude these members from the check.

**Comment by JoshuaKGoldberg** [comment]
Agreed, this isn't something general enough to TypeScript devs as a whole for us to add. It seems like a very useful proposal for a niche set of users. I'd recommend building a custom rule with the tooling documented in https://typescript-eslint.io/developers. Thanks for filing though!
